tardis/rest/service.py

```diff
@@ -0,0 +1,50 @@
+from .app.security import UserCredentials
+from cobald.daemon import service
+from cobald.daemon.plugins import yaml_tag
+
+from uvicorn.config import Config
+from uvicorn.server import Server
+
+from functools import lru_cache
+from typing import List
+import asyncio
+
+
+@service(flavour=asyncio)
+@yaml_tag(eager=True)
+class RestService(object):
+    def __init__(
+        self,
+        secret_key: str,
+        algorithm: str = "HS256",
+        users: List = None,
+        **fast_api_args,
+    ):
+        self._algorithm = algorithm
+        self._secret_key = secret_key
+        self._users = users or []
+
+        # necessary to avoid that the TARDIS' logger configuration is overwritten!
+        if "log_config" not in fast_api_args:
+            fast_api_args["log_config"] = None
+        self._config = Config("tardis.rest.app.main:app", **fast_api_args)
+
+    @property
+    @lru_cache(maxsize=16)
```

Comment:
Out of curiosity, as there was another discussion on the size of `lru_cache` before. Are we expecting to have several REST services that we now have a `maxsize` of `16`?

Reply:
It is not strictly enforced to have only one running. For example, I could imagine to have multiple REST services running with different scopes. One that permits read-only access listing the external interface, one with more access rights reachable only from the internal network. Not saying this is reasonable and a common use-case, but it is not forbidden.